cli/tools/task.rs

```diff
@@ -345,23 +345,31 @@ impl ShellCommand for NpmCommand {
   ) -> LocalBoxFuture<'static, ExecuteResult> {
     if context.args.first().map(|s| s.as_str()) == Some("run")
       && !context.args.iter().any(|s| s == "--")
+      && context.args.get(1).is_some()
     {
-      if let Some(task_name) = context.args.get(1) {
-        // run with deno task instead
-        let mut args = vec!["task".to_string(), task_name.to_string()];
-        args.extend(context.args.iter().skip(2).cloned());
-        let mut state = context.state;
-        state.apply_env_var(USE_PKG_JSON_HIDDEN_ENV_VAR_NAME, "1");
-        return ExecutableCommand::new(
-          "deno".to_string(),
-          std::env::current_exe().unwrap(),
-        )
-        .execute(ShellCommandContext {
-          args,
-          state,
-          ..context
-        });
+      // run with deno task instead
+      let mut args = vec!["task".to_string()];
+
+      for arg in context.args.iter().skip(1) {
+        // Skip arguments to `npm run` for now.
+        if arg.starts_with('-') {
```

Comment:
I think we can just skip if the first argument starts with a `-`. I'll update this PR to get it in the release.

Reply:
Oh wow, `npm run echo -s` will still turn on silent logging and not just pass that along? That's crazy.

Reply:
Crazy `npm run` doesn't pass along these. I think for now to be safe, let's change this to ignore if any argument starts with `-`. For example, I think it breaks atm for stuff like `--workspaces`. I'll update the pr.